src/ToggleButton.js

```diff
@@ -45,65 +50,74 @@ const propTypes = {
    * @type {ReactRef}
    */
   inputRef: PropTypes.any,
+
+  /**
+   * The defaultChecked state of the input
+   */
+  defaultChecked: PropTypes.bool,
 };
 
-const ToggleButton = React.forwardRef(
-  (
-    {
-      children,
-      name,
-      className,
-      checked,
-      type,
-      onChange,
-      value,
-      disabled,
-      inputRef,
-      ...props
-    },
-    ref,
-  ) => {
-    const [focused, setFocused] = useState(false);
-
-    const handleFocus = useCallback(e => {
-      if (e.target.tagName === 'INPUT') setFocused(true);
-    }, []);
-
-    const handleBlur = useCallback(e => {
-      if (e.target.tagName === 'INPUT') setFocused(false);
-    }, []);
-
-    return (
-      <Button
-        {...props}
-        ref={ref}
-        className={classNames(
-          className,
-          focused && 'focus',
-          disabled && 'disabled',
-        )}
-        type={null}
-        active={!!checked}
-        as="label"
-      >
-        <input
-          name={name}
-          type={type}
-          value={value}
-          ref={inputRef}
-          autoComplete="off"
-          checked={!!checked}
-          disabled={!!disabled}
-          onFocus={handleFocus}
-          onBlur={handleBlur}
-          onChange={onChange || noop}
-        />
-
-        {children}
-      </Button>
-    );
-  },
-);
+const ToggleButton = React.forwardRef((props, ref) => {
+  let {
+    children,
+    name,
+    className,
+    defaultChecked,
+    checked = defaultChecked,
+    type,
+    onChange,
+    value,
+    disabled,
+    inputRef,
+    ...controlledProps
+  } = useUncontrolled(props, {
+    checked: 'onChange',
+  });
+
+  const [focused, setFocused] = useState(false);
+
+  const handleFocus = useCallback(e => {
+    if (e.target.tagName === 'INPUT') setFocused(true);
+  }, []);
+
+  const handleBlur = useCallback(e => {
+    if (e.target.tagName === 'INPUT') setFocused(false);
+  }, []);
+
+  const handleChange = event => {
+    onChange(!checked, event);
+  };
```

Comment:
this would be a breaking change, as it would change the behavior of explicitly-configured `onChange` callbacks.

@jquense do we need to change the API here?

Reply:
I have reverted this so that we maintain the behaviour of the explicitly configured onChange callbacks. 
@taion

Reply:
does checking/unchecking still work when using `defaultChecked`, though?

Reply:
It works but the user would have to manually handle the checking/unchecking for the Toggle buttons.